packages/grid/_modules_/grid/components/columnSelection/GridHeaderCheckbox.tsx

```diff
@@ -42,13 +42,19 @@ const GridHeaderCheckbox = React.forwardRef<HTMLInputElement, GridColumnHeaderPa
       gridPaginatedVisibleSortedGridRowIdsSelector,
     );
 
-    const filteredSelection = React.useMemo(
-      () =>
-        typeof rootProps.isRowSelectable === 'function'
-          ? selection.filter((id) => rootProps.isRowSelectable!(apiRef.current.getRowParams(id)))
-          : selection,
-      [apiRef, rootProps.isRowSelectable, selection],
-    );
+    const filteredSelection = React.useMemo(() => {
+      if (typeof rootProps.isRowSelectable !== 'function') {
+        return selection;
+      }
+
+      return selection.filter((id) => {
+        try {
```

Comment:
What do you think about filtering `selection` to keep only the row ids that are visible in the current page. This would remove the need to use a try/catch.

In https://mui.com/components/data-grid/demo/ I can't accumulate ids to the previous selection. Gmail doesn't keep the checkbox state between pages, it always begins unchecked.

![msedge_S3Bp2ipmFJ](https://user-images.githubusercontent.com/42154031/149942511-a8fc829a-db11-4edc-915b-a3a8210a49cb.gif)

Reply:
We would loose the logic behind `checkboxSelectionVisibleOnly`
Right now users can decide if they want the select all to work across all pages or only on the current page
(`VisibleOnly` is wrongly named since "visible" doesn't mean "current page" right now though)

Reply:
Ok we can ignore this idea to not break `checkboxSelectionVisibleOnly`. However, I'm not a big fan of using a try/catch. It might hide things. If `isRowSelectable` or `apiRef.current.getRowParams` crashes for some reason the error will be swallowed and an error boundary won't catch it.

Reply:
Could we say in v6 that `getRowParams` return a nullable value instead of crashing ?
That way we can avoid `try / catch`
As this method is called a lot of times, I'm not a fan of adding a check before to know if the row do exist. But that would also by an alternative solution.